src/extensions/score_metamodel/__init__.py

```diff
@@ -133,6 +134,11 @@ def is_check_enabled(check: local_check_function | graph_check_function):
             logger.debug(f"Running local check {check} for need {need['id']}")
             check(app, need, log)
 
+    # External needs: run a focused, info-only check on optional_links patterns
+    # so that optional link issues from imported needs are visible but do not
+    # fail builds with -W.
+    _check_external_optional_link_patterns(app, log)
```

Comment:
Why add a check for external needs? They are checked as internal needs in their respective repos.